third_party/3/pyspark/sql/_typing.pyi

```diff
@@ -10,6 +10,7 @@ ColumnOrName = Union[pyspark.sql.column.Column, str]
 DecimalLiteral = decimal.Decimal
 DateTimeLiteral = Union[datetime.datetime, datetime.date]
 LiteralType = Union[bool, int, float, str]
+DataTypeOrString = Union[pyspark.sql.types.DataType, str]
```

Comment:
As far as I can tell that is too broad definition. For example it allows for `StructType` columns,

```
>>> from pyspark.sql.types import DataType, StructType                                                                                                                                   
>>> issubclass(StructType, DataType)                                                                                                                                                     
True
```

n contexts, where these are not allowed (current Pandas UDF's)

Reply:
That is true, but how would you suggest to fix that? According to the docs everything except for `MapType` and `StructType` is allowed as a return type for Pandas UDFs, should I list all other DataTypes explicitly?